tools/run_tests/run_tests.py

```diff
@@ -488,19 +490,35 @@ def _get_pythons(self, args):
       builder_prefix_arguments = []
       venv_relative_python = ['bin/python']
       toolchain = ['unix']
+
       # Bit-ness is handled by the test machine's environment
       python_pattern_function = lambda major, minor, bits: 'python{major}.{minor}'.format(major=major, minor=minor)
+
+      pypy_pattern_function = lambda major: 'pypy{major}'.format(major=major) if major == 3 else 'pypy'
+
     runner = [os.path.abspath('tools/run_tests/run_python.sh')]
+
     python_config_generator = lambda name, major, minor, bits: PythonConfig(
```

Comment:
Rather than further bloat this already-too-long method, pull both `python_config_generator` and `pypy_config_generator` out into module-scope module-private helper functions?